docs/DevGuide/BuildOptimization.md

```diff
@@ -208,6 +208,34 @@ It is recommended to install and use
 to use the full KDE window system) to visualize the output -- the larger graphs
 produced by templight are inefficient to render in the graphviz format.
 
+## Clang profiling
+
+With `clang >= 9`, you can generate a flame graph of what the compiler is
+spending its time on, including specific function instantiations, class
+instantiations, source file parsing, debug info generation, and function
+optimization. To do so, compile your `cpp` of interest with `clang >= 9` and use
+the additional flag `-ftime-trace`. You can find the full compilation command
+for your `cpp` of interest by searching for it in the `compile_commands.json`
+file in your `build` directory. This will produce a
+`<your_cpp_target_name>.cpp.json` that you can then load into
+`chrome://tracing`. Because our code base utilizes many templates and some
+compilation targets instantiate many types, classes, and functions, note that
+the flame graph for a particular compilation target may be a lot to visually
+navigate, click through, and analyze at a high level. Because of this, the tool
+described below can be very useful and more manageable for some analyses.
+
+The companion tool to the flame graph generation is the
+[ClangBuildAnalyzer](https://github.com/aras-p/ClangBuildAnalyzer). This is an
+open source tool that you can clone and run to give you profiling output
+similar to when you profile runtime code in that it identifies compilation hot
+spots for you. It will report things like "Here are the templates/functions
+that took the longest to instantiate/compile." Note that because many of our
+types can have very long names due to our templating, some class and function
+signatures will not fit in the default character limit of what is printed to the
+terminal. You can adjust the maximum character length printed by editing the
+`ClangBuildAnalyzer.ini` file as described in the
```

Comment:
by creating a `ClangBuildAnalyzer.ini` file in your working directory as described...